File: 07_Sorting/python syntax/04_Bubble_sort/bubble_sort.py
# Bubble sort

def bubble_sort(arr):
    for i in range(len(arr)-1):
        for j in range(len(arr)-1-i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                
    print(arr)
    

arr = [2,5,6,1,4,3]
bubble_sort(arr)

File: 07_Sorting/Cpp_syntax/02_Bubble_sort/Bubble_sort.cpp
#include <iostream>
using namespace std;

void Bubble_sort(int arr[], int size) {

    for (int i = 1; i < size; i++) {

        bool swapped = false;

        for (int j = 0; j < size - i; j++) {

            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
                swapped = true;
            }
        }

        // Stop if array is already sorted
        if (!swapped) {
            break;
        }
    }
}

void print(int arr[], int size) {

    for (int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }

    cout << endl;
}

int main() {

    int arr[10] = {3, 5, 21, 6, 4, 9, 8, 7, 10, 2};
    int size = 10;

    Bubble_sort(arr, size);
    print(arr, size);

    return 0;
}